tests/parser/UDQTests.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright 2018 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE UDQTests
+#include <boost/test/unit_test.hpp>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQ.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQRecord.hpp>
+
+using namespace Opm;
+
+BOOST_AUTO_TEST_CASE(KEYWORDS) {
+    const std::string input = R"(
+    RUNSPEC
```

Comment:
Do note that if you write it like this, then the leading blanks on each line become part of the string.  In other words, this is strictly speaking not valid ECL input (which requires that all keywords start in column 1).

You can verify by running the example below
```C++
#include <iostream>
#include <string>

int main() {
    const auto s = std::string { R"(
        Hi there
    )" };

    std::cout << s;
} 
```

Reply:
Thank you - will fix!